glacier/src/reducers/fields.ts

```diff
@@ -0,0 +1,23 @@
+import {FieldState, Field} from "../model";
+import {AllActions} from "../actions";
+
+export function fields(state: FieldState | undefined, action: AllActions) {
+     if (!state) return {};
+     if (action.error) {
+         throw action.error;
+     }
+     switch (action.type) {
+         case "ADD_FIELD": {
+            return {...state, ...action.payload};
```

Comment:
`action.payload` is an array (as is `state`) - while this is valid, I'm pretty sure you don't mean to spread `[{a}, {b}, {c}]` into `{'0': {a}, '1': {b}, '2': {c}}`. You probably want `[...state, ...action.payload]`.